### ISSUE TITLE
[Google Shopping API] Missing `inline_shopping_results` for `device=desktop`

### ISSUE THREAD

**Issue by NateSkiles** [serpapi/public-roadmap#2154]
A customer reached out to report they were no longer seeing `inline_shopping_results`. This appears to only be affecting `desktop` searches, but `inline_shopping_results` are missing from the HTML of all devices.

### Desktop

![image](https://github.com/user-attachments/assets/ab5b3681-7a12-4265-9952-08c96cdaa913)

[Playground](https://serpapi.com/playground?engine=google_shopping&q=ps5+controller&location=Portland%2C+Oregon%2C+United+States&gl=us&hl=en&device=desktop&newPara=direct_link) | [Inspect](https://serpapi.com/searches/673d0aa4d78a984658b247c2/inspect)

![image](https://github.com/user-attachments/assets/65f967f9-2a5c-4a7d-86ff-d0059e319026)

https://www.google.com/search?udm=28&q=ps5+controller&uule=w+CAIQICIdUG9ydGxhbmQsT3JlZ29uLFVuaXRlZCBTdGF0ZXM&hl=en&gl=us

### Mobile / Tablet

![image](https://github.com/user-attachments/assets/b1d52a6a-b8e0-4e05-a6b8-2771c1737a85)

[Playground](https://serpapi.com/playground?engine=google_shopping&q=ps5+controller&location=Portland%2C+Oregon%2C+United+States&gl=us&hl=en&device=mobile&no_cache=true&newPara=direct_link) | [Inspect](https://serpapi.com/searches/673d3460b5ecbba0649e2920/inspect)

![image](https://github.com/user-attachments/assets/be48a316-219f-48bd-8d44-ff160d4f192e)

[Google Shopping URL](https://www.google.com/search?udm=28&q=ps5+controller&uule=w+CAIQICIdUG9ydGxhbmQsT3JlZ29uLFVuaXRlZCBTdGF0ZXM&hl=en&gl=us)

[Documentation](https://serpapi.com/google-shopping-new-layout)

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000072308?view=List)

**Comment by btaunt** [comment]
Hey @NateSkiles 

I'm having a difficult time replicating this issue. Even the provided `inspect` for the desktop variation shows the `inline_shopping_results` both in the rendered HTML and JSON output. Unless I am misunderstanding the reported problem?

<img width="1177" alt="Screenshot 2024-11-26 at 2 30 54 PM" src="https://github.com/user-attachments/assets/82e8d84e-3cc1-430e-a293-719ae1bbcda6">

**Comment by schaferyan** [comment]
The inspect link for the Desktop search doesn't seem to match the Playground screenshot. @NateSkiles is it possible you copied the wrong Inspect link?

When retrying the same search several times, the `inline_shopping_results` are sometimes present and sometimes not. However, I believe that inconsistency is on Google's side, so if that's what the customer is referring to, I unfortunately don't think we have any ability to affect that.

**Comment by fz-phia** [comment]
Hi all, I tried many keywords. However, none of them return the inline_shopping_results,
for example, this is the query I used "used Eternity solitaire ring Swarovski White". I could see the sponsored products from google shopping on the webpage. However, it's never returned in the google serp API

**Comment by fz-phia** [comment]
<img width="1710" alt="image" src="https://github.com/user-attachments/assets/9c0bb264-a9a5-4506-853e-ffe873ab4fc0">
This is the ps5 controller from the playground. No sponsored products

**Comment by schaferyan** [comment]
Hi @fz-phia - thanks for the additional information. Can you please try resubmitting the "ps5 controller" query in the playground 5-10 times with `no_cache` set to `true`? You should see the `inline_shopping_results` for some of the retries.

My understanding is that Google Shopping only includes the sponsored results intermittently, as they use many factors to determine whether to show these. Unfortunately, if Google doesn't always show this result type for all similar searches, there's no way for us to guarantee they will always be present.

I'll leave this issue open for now, just in case there is anything we can do to influence this behavior.

I hope that makes sense, and thanks for your patience.

**Comment by NateSkiles** [comment]
@schaferyan @btaunt I had only tested this in the playground (with `no_cache`) and hadn't received `inline_shopping_results` for any desktop results. This was likely just a coincidence, as I ran a script checking for `inline_shopping_results` and found it was returned for 41.0% of the requests (out of 500).

Oddly, when setting `device=mobile` the success rate jumped to 97.6% and 79.4% for `tablet`.

Anecdotally, when testing this across multiple browsers (in incognito), the `inline_shopping_results` rarely returned when I first loaded the page but were always there after a refresh. Perhaps Google is loading these in at some point.

https://www.google.com/search?udm=28&q=ps5+controller&uule=w+CAIQICIdUG9ydGxhbmQsT3JlZ29uLFVuaXRlZCBTdGF0ZXM&hl=en&gl=us

This might be worth investigating further; however, it does appear to be an issue on Google's end.

@fz-phia As I mentioned above, I found searching for `device=mobile` and `device=tablet` returns `inline_shopping_results` much more consistently. While I'm unsure of your use case, I recommend testing this on your end and seeing if this is a good workaround.

**Comment by sonika-serpapi** [comment]
A user reached out about this: 
[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000075526?view=List)

**Comment by alexbarron** [comment]
@fz-phia followed up with us to report he now cannot get `inline_shopping_results` regardless of `device` setting.

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000075807?view=List)

**Comment by btaunt** [comment]
@alexbarron Just for reference, leaving `location` blank and searching via `mobile` device consistently returns `inline_shopping_results` for me. Not sure if this is a US location-based issue?

[Playground Example](https://serpapi.com/playground?engine=google_shopping&q=used+Eternity+solitaire+ring+Swarovski+White&gl=us&hl=en&device=mobile&no_cache=true&newPara=direct_link)

<img width="735" alt="Image" src="https://github.com/user-attachments/assets/dce0033e-4b0d-443b-8ee0-46d20fefdc9d" />

**Comment by fz-phia** [comment]
Yes, this is US based issue. I set the device as “tablet”, with location
New York. Without location, many times, there’s no sponsored products. Even
with it, i don’t get it constantly

**Comment by hilmanski** [comment]
Reports:

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/215470920268607?view=List)

**Comment by hilmanski** [comment]
A high-volume customer reported this.

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/215471653963586?view=List)

**Comment by jbcoventry** [comment]
A customer reported this:

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/215471686634612#part_id=initial-part-215471686634612-3255693688)

**Comment by schaferyan** [comment]
Another customer reported this:

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/215471957318232?view=List)

**Comment by jbcoventry** [comment]
A user reported this:

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/215472079055813)

**Comment by hilmanski** [comment]
[User reports tracking](https://docs.google.com/spreadsheets/d/107lQz0XNQkBlleJxkwpZllh1PkCTNtoWmiEfP38eKwE/edit?gid=0#gid=0)
